File: components/reference/manual-examples/tutorial-wiki/component/applications/wiki/resources/api.m.js
document.require(
	'/prudence/resources/',
	'/sincerity/json/')

function handleInit(conversation) {
	conversation.addMediaTypeByName('application/json')
	var id = 'page.' + conversation.wildcard
	caching.duration = 60000
	caching.tags.add(id)
}

function handleGet(conversation) {
	var id = 'page.' + conversation.wildcard
	var content = application.globals.get(id) || 'This page is empty.'
	return Sincerity.JSON.to({content: content})
}

function handlePut(conversation) {
	var id = 'page.' + conversation.wildcard
	var payload = Prudence.Resources.getEntity(conversation, 'json')
	application.globals.put(id, payload.content)
	document.cache.invalidate(id)
	return Sincerity.JSON.to({content: payload.content})
}

function handleDelete(conversation) {
	var id = 'page.' + conversation.wildcard
	application.globals.remove(id)
	document.cache.invalidate(id)
	return null
}

File: components/reference/manual-examples/tutorial-cms-python/component/applications/wiki/resources/api.m.py
import simplejson

def handle_init(conversation):
    conversation.addMediaTypeByName('application/json')
    id = 'page.' + conversation.wildcard
    caching.duration = 60000
    caching.tags.add(id) 

def handle_get(conversation):
    id = 'page.' + conversation.wildcard
    content = application.globals[id] or 'This page is empty.'
    return simplejson.dumps({'content': content})

def handle_put(conversation):
    id = 'page.' + conversation.wildcard
    payload = simplejson.loads(conversation.entity.text)
    application.globals[id] = payload['content']
    document.cache.invalidate(id)
    return simplejson.dumps({'content': payload['content']})

def handle_delete(conversation):
    id = 'page.' + conversation.wildcard
    del application.globals[id]
    document.cache.invalidate(id)
    return None
